aspnetcore/fundamentals/middleware/request-response.md

```diff
@@ -70,6 +70,11 @@ These issues are fixable, but the code is becoming progressively more complicate
 
 ## Pipelines
 
+When writing directly to `HttpResponse.BodyWriter`, call `PipeWriter.FlushAsync()` manually to ensure the data is flushed to the underlying response stream. Here's why:
+
+* `HttpResponse.BodyWriter` is a `PipeWriter` that buffers data until a flush operation is triggered.
+* Calling `FlushAsync` writes the buffered data to the underlying Stream (`HttpResponse.Body`).
```

Comment:
Should we specify how frequently the FlushAsync should be called? Once? Once per Advance? Every X bytes written?

Reply:
> Should we specify how frequently the FlushAsync should be called? Once? Once per Advance? Every X bytes advanced?

@gfoidl @martincostello ???

Reply:
> My guess is it's hard to give a simple answer.

True. A generic rule when to call `Flush` is hard to impossible to give.

Very general:
* call `Flush` to make the data available to the `PipeReader` (here: the response body), so that the response can be transmitted (chunked) over the wire to the client
* call `Flush` when someting is written that the `PipeReader` should have access to (before completing the pipe)

But I don't know how to defer a rule out of the points.
Honestly, I'd just omit something like this, as at the end of the article there's a link to _System.IO.Pipelines in .NET_ which has further infos.

Reply:
Could we say something like "It's up to the developer to decide when to call FlushAsync, balancing factors such as buffer size, network write overhead, and whether the data should be sent in discrete chunks."